static/app/views/dashboards/manage/index.spec.tsx

```diff
@@ -44,59 +52,163 @@ describe('Dashboards > Detail', function () {
+    MockApiClient.addMockResponse({
+      url: '/organizations/org-slug/dashboards/',
+      statusCode: 400,
+    });
+    render(<ManageDashboards />, {
+      ...RouteComponentPropsFixture(),
+      organization: mockAuthorizedOrg,
+    });
+
+    expect(await screen.findByText('Oops! Something went wrong')).toBeInTheDocument();
+  });
+
+  it('denies access on missing feature', async function () {
+    render(<ManageDashboards />, {
+      ...RouteComponentPropsFixture(),
+      organization: mockUnauthorizedOrg,
+    });
 
-    expect(screen.getByText("You don't have access to this feature")).toBeInTheDocument();
+    expect(
+      await screen.findByText("You don't have access to this feature")
+    ).toBeInTheDocument();
   });
 
-  it('denies access on no projects', function () {
+  it('denies access on no projects', async function () {
     act(() => ProjectsStore.loadInitialData([]));
 
-    render(
-      <ManageDashboards
-        {...RouteComponentPropsFixture()}
-        organization={mockAuthorizedOrg}
-      />
-    );
+    render(<ManageDashboards />, {
+      ...RouteComponentPropsFixture(),
+      organization: mockAuthorizedOrg,
+    });
 
     expect(
-      screen.getByText('You need at least one project to use this view')
+      await screen.findByText('You need at least one project to use this view')
     ).toBeInTheDocument();
   });
 
   it('creates new dashboard', async function () {
     const org = OrganizationFixture({features: FEATURES});
+    const mockNavigate = jest.fn();
+    mockUseNavigate.mockReturnValue(mockNavigate);
 
-    render(<ManageDashboards {...RouteComponentPropsFixture()} organization={org} />);
+    render(<ManageDashboards />, {
+      ...RouteComponentPropsFixture(),
+      organization: org,
+    });
 
-    await userEvent.click(screen.getByTestId('dashboard-create'));
+    await userEvent.click(await screen.findByTestId('dashboard-create'));
 
-    expect(browserHistory.push).toHaveBeenCalledWith({
+    expect(mockNavigate).toHaveBeenCalledWith({
       pathname: '/organizations/org-slug/dashboards/new/',
       query: {},
     });
   });
 
   it('can sort', async function () {
     const org = OrganizationFixture({features: FEATURES});
+    const mockNavigate = jest.fn();
+    mockUseNavigate.mockReturnValue(mockNavigate);
 
-    render(<ManageDashboards {...RouteComponentPropsFixture()} organization={org} />);
+    render(<ManageDashboards />, {
+      ...RouteComponentPropsFixture(),
+      organization: org,
+    });
 
     await selectEvent.select(
-      screen.getByRole('button', {name: /sort by/i}),
+      await screen.findByRole('button', {name: /sort by/i}),
       'Dashboard Name (A-Z)'
     );
 
-    await waitFor(() => {
-      expect(browserHistory.push).toHaveBeenCalledWith(
-        expect.objectContaining({query: {sort: 'title'}})
-      );
+    expect(mockNavigate).toHaveBeenCalledWith(
+      await expect.objectContaining({query: {sort: 'title'}})
```

Comment:
I don't think expect is async so awaiting doesn't do anything, you can probably remove these